packages/datadog-plugin-dns/src/index.js

```diff
@@ -44,7 +44,8 @@ class DNSPlugin extends Plugin {
       this.enter(span, store)
     }, (result) => {
       const { span } = storage.getStore()
-      span.setTag('dns.address', result)
+      const address = Array.isArray(result) ? result[0].address : result
+      span.setTag('dns.address', address)
```

Comment:
If I'm reading this right, would it not be better in the case of an array to do something like `result.map((r) => r.address).join(', ')` to ensure all resolved addresses are listed?
Maybe a bit misleading if you expect `dns.address` to include 1 ip, but I think now its misleading in another way saying a DNS only resolved 1 address when in fact it would have resolved to several of them?

Reply:
The default behaviour without `all` is to take the first one, so I went with the same approach to keep a single address. I'm not sure what makes the most sense since I'm not sure what the point of the option is in the first place and in what scenario the first one is not what would be used 😅 Do you know why `grpc` uses the `all` option?

Reply:
We for various reasons use client side load balancing on our gRPC setup, so I believe that gRPC uses the all option to resolve all the potential IPs for the domain so it can do round robin on the IPs it resolved.
But this is a guesstimate from my side 😄 

Yesterday we had these traces during an incident 
![image](https://user-images.githubusercontent.com/326929/166832315-85d0690c-1c16-4a6d-bfa9-f451a7b16678.png)

As can be seen it performs a dns lookup and then proceeds to try to connect to each of the 3 resolved IPs (or at least I assume those IPs would match what the dns lookup resulted in)

Reply:
How do you use the tag? Would it make more sense to split the address as multiple tags or comma-separated is fine? I wonder if it should go under a different name too.

Reply:
Human observation 😄 the automatic dns instrumentation has helped us in a few cases when reading traces.
I think either solution would be fine, in the case of an array putting it under `addresses` as a comma separated list could make sense.